### ISSUE TITLE
is it necessary to finetune the t5 model?

### ISSUE THREAD

**Issue by Tresfroid** [zhouj8553/FlipDA#4]
the results are not correct when i generate augment examples using t5 chinese model, is it necessary to finetune the t5 model?

**Comment by zhouj8553** [comment]
Sorry to hear that. We do not fine-tune the T5 model. (It will undoubtedly overfit the training set in the few-shot scenario).

I didn't try the Chinese T5 model, but I have the following suggestions, which might be helpful. Maybe you could judge what kind of situation you are in and then choose the corresponding actions.

1. The generated sentences are not fluent or even wrong in grammar. It should be solved almost perfectly because the pretrained models are powerful. In this case, (1) you should check whether the fill-in function is correct, and (2) you should check whether the model is good enough. (For example, T5-large is much much better than T5-base.)

2. The generated sentences look good but with the wrong labels. In this case, the following actions might help.
(1) generate a lot of candidates and then use the classifier to find out correct examples (as we mentioned as self-training). It will still not be totally correct, but some wrong labeled examples will not influence the performance a lot. 
(2) Sometimes the task is too difficult. You could try to construct the prompt in another way and make the model predict more easily. (Well, it is flexible according to your problem) 
(3) I didn't try this or check the details. I think the idea might help if you still think finetune is necessary. Treat the generating process also a few-shot learning process, and use prompt-based methods to finetune the T5-model (a little like Paper "PromDA: Prompt-based Data Augmentation for Low-Resource NLU Tasks" does.) 
(4) The last way that I could think of is to split the data into two parts, i.e., A and B. Finetune part A and predict for part B, and then finetune part B and then predict for part A. (It might mitigate overfitting to some extent.)
